src/state_point.F90

```diff
@@ -866,14 +790,12 @@ subroutine load_state_point()
       ! Read in sum and sum squared
       if (int_array(1) == 1) then
         TALLY_RESULTS: do i = 1, n_tallies
-
           ! Set pointer to tally
           tally => tallies(i)
-          curr_key = key_array(id_array(i))
 
           ! Read sum and sum_sq for each bin
           tally_group = open_group(tallies_group, "tally " // &
-               trim(to_str(curr_key)))
```

Comment:
How does this error out if the user adds a new tally to their "tallies.xml" file? Or in the reverse scenario if a tally has been removed from the "tallies.xml" file, no error will be reported and this tally will simply not be reported in later statepoints. What do you think about some sort of error checking capability here to deal with this?